### ISSUE TITLE
Ubuntu 22.04 clang build broken: <chrono> header undefined function

### ISSUE THREAD

**Issue by geoffthemedio** [freeorion/freeorion#4774]
https://github.com/freeorion/freeorion/actions/runs/6651865453/job/18074754759#step:11:96

````
In file included from /home/runner/work/freeorion/freeorion/parse/../util/ScopedTimer.h:10:
/usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/chrono:2320:48: error: call to consteval function 'std::chrono::hh_mm_ss::_S_fractional_width' is not a constant expression
        static constexpr unsigned fractional_width = {_S_fractional_width()};
                                                      ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/chrono:2320:48: note: undefined function '_S_fractional_width' cannot be used in a constant expression
/usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/chrono:2275:2: note: declared here
        _S_fractional_width()
        ^
````

possibly is related to:
https://github.com/actions/runner-images/issues/8659

and the error is reported elsewhere, eg:
https://www.mail-archive.com/debian-bugs-dist@lists.debian.org/msg1918839.html
https://stackoverflow.com/questions/76859275/error-compiling-a-cpp-containing-stdchrono-errorstatic-constexpr-unsigned-fra
https://sources.debian.org/src/chromium/118.0.5993.70-1/debian/patches/fixes/gcc13-with-clang14.patch/

@o01eg is there something we can do to work around this? I don't see an easy way to fix it in code... maybe disabling all inclusion of the `<chrono>` header on clang, but I'm not sure how to check what versions of it or standard libraries are affected. Can `CMakeLists.txt` settings be modified to specify a different standard library for clang?

**Comment by geoffthemedio** [comment]
I'm trying updating / using a newer clang in #4775 but that's not ideal if the default in Ubuntu 22.04 is something lower. Not sure how much this issue is specific to GitHub's runner's particular versions. vs. what anyone using Ubuntu would have by default, though... The tests are less useful if they don't reproduce a standard setup.

**Comment by o01eg** [comment]
Not sure be we can disable clang build in CI and make check in CMakeLists.txt for never clang.

**Comment by geoffthemedio** [comment]
I guess I'd rather use clang-15 than not at all...

**Comment by geoffthemedio** [comment]
seeminlg resolved by 4b34756471
